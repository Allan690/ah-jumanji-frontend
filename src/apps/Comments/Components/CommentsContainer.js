// Container for all comments. Will be imported into Articles app

import React, {Component} from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';

import CommentsNav from './CommentsNav';
import Comment from './Comment';
import InfoSubscript from './Reusable/InfoSubscript';
import AddCommentComponent from './AddComment';
import {fetchComments} from '../Actions/CommentActions';
import LikesViewContainer from "../../Like/containers/LikesViewContainer";

import { fetchCommentReactions } from '../../UserReactions/Actions/CommentRxnActions';


class CommentsContainer extends Component {
  componentWillMount() {
    const {fetchComments, fetchCommentReactions} = this.props;
    fetchCommentReactions();
    setTimeout(() => {
      fetchComments();
    }, 1000);
  }

  render() {
    const {comments, likeCount, reactions} = this.props;
    return (
      <div>
        <CommentsNav numberOfComments={comments.length} numberOfLikes={likeCount} />
        <div className="tab-content" id="myTabContent">
          <div className="tab-pane fade show active" id="comments" role="tabpanel" aria-labelledby="tab">
            <div className="col-md-12">
              <ul id="comments-list" className="list-group list-group-flush">
                {/* Render list of comments. Or message if no comments */}
                {
                  comments.length ? comments.map(comm => (
                    <Comment
                      key={comm.id}
                      comment={comm}
                      reactions={reactions.filter(rxn => rxn.comment === comm.id)}
                    />
                  )) : <InfoSubscript extraClasses="hidden" text="No comments on this one. Yet." />
                }
                <AddCommentComponent actionToExecute="addNew" specialClass="unnecessary" />
              </ul>
            </div>
          </div>
          <LikesViewContainer />
        </div>
      </div>
    );
  }
}

CommentsContainer.propTypes = {
  comments: PropTypes.array.isRequired,
  fetchComments: PropTypes.func.isRequired,
  fetchCommentReactions: PropTypes.func.isRequired,
  likeCount: PropTypes.number.isRequired,
  reactions: PropTypes.array.isRequired,
};

const mapStateToProps = (state) => ({
  comments: state.commentListReducer.comments,
  likeCount: state.LikeReducer.likeCount,
  reactions: state.commentsReactionsReducer.reactions
});

export default connect(
  mapStateToProps, {fetchComments, fetchCommentReactions})(CommentsContainer);
