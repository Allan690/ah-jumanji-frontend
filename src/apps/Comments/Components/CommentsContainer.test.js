import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import CommentsContainer from "./CommentsContainer";

Enzyme.configure({adapter: new Adapter()});

const fakeComment = {
  "id": 100,
  "author": {
    "username": "dmithamo",
    "image": "",
  },
  "body": "fake body"
};

const fakeCommentToo = {
  "id": 1000,
  "author": {
    "username": "wewe",
    "image": "",
  },
  "body": "Thats what I said"
};

const fakeCommentThree = {
  "id": 10000,
  "author": {
    "username": "mr storys",
    "image": "",
  },
  "body": "One more story"
};

const commentList = {"comments": [fakeComment, fakeCommentToo, fakeCommentThree]};

describe('<CommentsContainer />', () => {
  it('render all expected pieces', () => {
    const component = shallow(<CommentsContainer comments={commentList} />);
    expect(component).toMatchSnapshot();
  });
});
