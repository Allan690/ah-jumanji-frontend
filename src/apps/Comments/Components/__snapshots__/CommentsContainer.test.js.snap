// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentsContainer /> render all expected pieces 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
